src/__tests__/components/CreateTemplate/index.test.tsx

```diff
@@ -0,0 +1,74 @@
+    const previewTitle = screen.getByText("Court Preview:");
+    const nameTitle = screen.getByText("Template Court Name:");
+    const descriptionTitle = screen.getByText("Description:");
+    const publisherTitle = screen.getByText("Publisher:");
+
+    expect(modalTitle).toBeInTheDocument();
+    expect(previewTitle).toBeInTheDocument();
+    expect(nameTitle).toBeInTheDocument();
+    expect(descriptionTitle).toBeInTheDocument();
+    expect(publisherTitle).toBeInTheDocument();
+  });
+
+  it("Should render buttons in modal footer", () => {
+    renderWithMockedProvider(<CreateTemplate isOpen={true} onClose={() => void {}} />);
+
+    const publishBtn = screen.getByRole("button", { name: "Publish" });
+    const cancelBtn = screen.getByRole("button", { name: "Cancel" });
+
+    expect(publishBtn).toBeInTheDocument();
+    expect(cancelBtn).toBeInTheDocument();
+  });
+
+  it("Should show error massage when court name input reaches the maximum length or court name input is empty", async () => {
+    renderWithMockedProvider(<CreateTemplate isOpen={true} onClose={() => void {}} />);
+    const courtNameInput = screen.getByRole("textbox", { name: "courtNameInput" });
+    fireEvent.change(courtNameInput, { target: { value: "12345678901234567890" } });
+    await waitFor(() => {
+      expect(screen.getByText("Court name cannot have more than 15 characters")).toBeVisible();
+    });
+    fireEvent.change(courtNameInput, { target: { value: "" } });
+    await waitFor(() => {
+      expect(screen.getByText("Court name cannot be empty")).toBeVisible();
+    });
+  });
+
+  it("Should show error message when click punish button and court name is empty", async () => {
+    renderWithMockedProvider(<CreateTemplate isOpen={true} onClose={() => void {}} />);
+    const publishBtn = screen.getByRole("button", { name: "Publish" });
+    userEvent.click(publishBtn);
+    await waitFor(() => {
+      expect(screen.getByText("Court name cannot be empty")).toBeVisible();
+    });
+  });
+
+  it("Error message should disappear when court name length is not empty", async () => {
+    renderWithMockedProvider(<CreateTemplate isOpen={true} onClose={() => void {}} />);
+    const courtNameInput = screen.getByRole("textbox", { name: "courtNameInput" });
+    fireEvent.change(courtNameInput, { target: { value: "1" } });
+    await waitFor(() => {
+      expect(screen.getByText("Court name cannot be empty")).not.toBeVisible();
+    });
+  });
+
+  it("The description word count should correctly count the input description length", async () => {
+    renderWithMockedProvider(<CreateTemplate isOpen={true} onClose={() => void {}} />);
+    const descriptionInput = screen.getByRole("textbox", { name: "textArea" });
+    fireEvent.change(descriptionInput, {
+      target: { value: "This my bag" },
+    });
+    await waitFor(() => {
+      expect(screen.getByText("11/200 letters")).toBeVisible();
+    });
+  });
+});
```

Comment:
I see that both fireEvent and userEvent are used here, but both of them have click and input methods. I recommend using userEvent
[common-mistakes-with-react-testing-library](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library)
<img width="1139" alt="image" src="https://user-images.githubusercontent.com/61776449/197171988-d52b90e5-e3dc-4ab3-8c1d-05cc60bb818a.png">

Reply:
All fireEvent has been changed to userEvent.